src/core/sock.c

```diff
@@ -1088,3 +1161,24 @@ void nn_sock_stat_increment (struct nn_sock *self, int name, int64_t increment)
             break;
     }
 }
+
+int nn_sock_hold (struct nn_sock *self)
+{
+    switch (self->state) {
+    case NN_SOCK_STATE_ACTIVE:
+    case NN_SOCK_STATE_INIT:
+        self->holds++;
+        return 0;
+    case NN_SOCK_STATE_ZOMBIE:
+        return -ETERM;
+    }
```

Comment:
There's not much the compiler can do for us here, going on pure C convention with no typedefs, but I find myself wondering ... "what other cases are there??"

Said another way ... it's hard to read this code and feel certain it's not wrong. It cannot explicitly tell me it's not wrong. Therefore, it's hard to reason whether it's correct.